lib/rule-data-snapshot.js

```diff
@@ -68,7 +68,29 @@ RuleDataSnapshot.prototype.merge = function(other) {
     throw new Error('can only merge top-level RuleDataSnapshots');
   }
 
-  var data = extend(true, {}, this._data, other._data);
+  var data = merge({}, this._data);
+
+  data = merge(data, other._data, function(oldNode, newNode, key) {
+    var oldNodeIsLiteral = oldNode && oldNode['.value'] !== undefined;
+    var newNodeIsLiteral = newNode && newNode['.value'] !== undefined;
+    var priority;
+
+    if (!oldNodeIsLiteral && !newNodeIsLiteral) {
+      return undefined;
```

Comment:
It's probably not obvious than returning `undefined` let `lodash.mergeWith` merge the two object.

I can have something like:
```js
const DEFAULT_MERGE = undefined;` and
[...]
if (!oldNodeIsLiteral && !newNodeIsLiteral) {
  return DEFAULT_MERGE;
}
```